client/app/pages/queries/QuerySource.jsx

```diff
@@ -348,36 +322,30 @@ function QuerySource(props) {
 
                   <QueryEditor.Controls
                     addParameterButtonProps={{
-                      title: (
-                        <React.Fragment>
-                          Add New Parameter (<i>{modKey} + P</i>)
-                        </React.Fragment>
-                      ),
+                      title: "Add New Parameter",
+                      shortcut: "mod+p",
                       onClick: openAddNewParameterDialog,
                     }}
                     formatButtonProps={{
-                      title: (
-                        <React.Fragment>
-                          Format Query (<i>{modKey} + Shift + F</i>)
-                        </React.Fragment>
-                      ),
+                      title: "Format Query",
+                      shortcut: "mod+shift+f",
                       onClick: formatQuery,
                     }}
                     saveButtonProps={
                       query.can_edit && {
-                        title: `${modKey} + S`,
                         text: (
                           <React.Fragment>
                             <span className="hidden-xs">Save</span>
                             {isDirty ? "*" : null}
                           </React.Fragment>
                         ),
+                        shortcut: "mod+s",
                         onClick: saveQuery,
                       }
                     }
                     executeButtonProps={{
-                      title: `${modKey} + Enter`,
-                      disabled: isEmpty(query.query) || !canExecuteQuery || isQueryExecuting,
+                      disabled: !canExecuteQuery,
+                      shortcut: "mod+enter, alt+enter",
```

Comment:
Very interesting shortcut definition :rocket:

Only thing is that for the Query execution button I think it's better if the tooltip shows only "[Mod]+Enter" (even though it also works for Alt) - looks too much with both shortcuts appearing in the tooltip, so perhaps make the tooltip replaceable if you agree?

Reply:
I decided that it may be useful to see all shortcuts, but I agree with your point - it's too crowded. Fixed in https://github.com/getredash/redash/pull/4468/commits/57bf86459f7fe8b3de79f79d2f8e56b7e4d7f9c8